props/saber_fett263_buttons.h

```diff
@@ -381,160 +1813,2794 @@ SaberFett263Buttons() : PropBase() {}
+                PlayMenuSound("mexit.wav");
+              }
+              wav_player.Free();
+              break;
+            case MENU_PRESET:
+              menu_ = false;
+              menu_type_ = MENU_TOP;
+              PlayMenuSound("mselect.wav");
+              wav_player.Free();
+              break;
+            default:
+              wav_player.Free();
+          }
         } else {
-          On();
+#ifdef FETT263_DUAL_MODE_SOUND
+          SelectOut();
+#endif          
+          if (SFX_preon) {
+            On();
+          } else {
+            FastOn();
+          }
 #ifdef FETT263_BATTLE_MODE_ALWAYS_ON
           battle_mode_ = true;
 #endif
 #ifdef FETT263_BATTLE_MODE_START_ON
           battle_mode_ = true;
 #endif
-        }
-        return true;
-
-#ifdef BLADE_DETECT_PIN
-      case EVENTID(BUTTON_BLADE_DETECT, EVENT_LATCH_ON, MODE_ANY_BUTTON | MODE_ON):
-      case EVENTID(BUTTON_BLADE_DETECT, EVENT_LATCH_ON, MODE_ANY_BUTTON | MODE_OFF):
-        // Might need to do something cleaner, but let's try this for now.
-        blade_detected_ = true;
-        FindBladeAgain();
-        SaberBase::DoBladeDetect(true);
-        return true;
+        }
+        return true;
+
+#ifdef BLADE_DETECT_PIN
+      case EVENTID(BUTTON_BLADE_DETECT, EVENT_LATCH_ON, MODE_ANY_BUTTON | MODE_ON):
+      case EVENTID(BUTTON_BLADE_DETECT, EVENT_LATCH_ON, MODE_ANY_BUTTON | MODE_OFF):
+        // Might need to do something cleaner, but let's try this for now.
+        blade_detected_ = true;
+#ifdef FETT263_SAVE_GESTURE_OFF    
+        RestoreGestureState();
+#else
+        saved_gesture_control.gestureon = true;
+#endif
+        FindBladeAgain();
+        SaberBase::DoBladeDetect(true);
+        return true;
+
+      case EVENTID(BUTTON_BLADE_DETECT, EVENT_LATCH_OFF, MODE_ANY_BUTTON | MODE_ON):
+      case EVENTID(BUTTON_BLADE_DETECT, EVENT_LATCH_OFF, MODE_ANY_BUTTON | MODE_OFF):
+        // Might need to do something cleaner, but let's try this for now.
+        blade_detected_ = false;
+#ifdef FETT263_SAVE_GESTURE_OFF    
+        SaveGestureState();
+#endif
+        saved_gesture_control.gestureon = false;
+        FindBladeAgain();
+        SaberBase::DoBladeDetect(false);
+        return true;
+#endif
+
+      case EVENTID(BUTTON_AUX, EVENT_CLICK_SHORT, MODE_OFF | BUTTON_POWER):
+        if (!menu_) {
+          current_menu_angle_ = fusor.angle2();
+          menu_ = true;
+          menu_type_ = MENU_VOLUME;
+          if (SFX_vmbegin) {
+            PlayMenuSound("vmbegin.wav");
+          } else {
+            PlayMenuSound("mvolume.wav");
+          }
+        }
+        return true;
+          
+      case EVENTID(BUTTON_AUX, EVENT_CLICK_LONG, MODE_OFF):
+        current_menu_angle_ = fusor.angle2();
+        PlayMenuSound("mpreset.wav");
+        menu_ = true;
+        menu_type_ = MENU_PRESET;
+        return true;
+
+      case EVENTID(BUTTON_AUX, EVENT_CLICK_SHORT, MODE_OFF):          
+        if (menu_) {        
```

Comment:
Add some comments, what does all this *do*?

Reply:
comment added